spynl/main/exceptions.py

```diff
@@ -19,17 +20,26 @@ class SpynlException(Exception):
     param developer_message: A message for 3rd party users of our API.
                              Exposed, should NOT contain sensitive data.
     param debug_message: A message for internal use when debugging.
+
+    External exceptions can be mapped to internal SpynlExceptions by registering
+    the external error with the 'this_exception_maps' decorator at the internal
+    SpynlException. For the mapping to work, the catch_mapped_exceptions
+    function needs to be registered as a view deriver.
     """
     http_escalate_as = HTTPBadRequest
 
     def __init__(self,
                  message='an internal error has occured',
                  developer_message=None,
                  debug_message=None):
+        # set messages
         self.message = message
         self.developer_message = developer_message
         self.debug_message = debug_message
 
+        # store a reference to original exception in case of mapping
+        self.original_exception = sys.exc_info()[1]
```

Comment:
I really do not think this is a bad idea. This might lead to all sort of side effects. E.g. imagine something like this:

```
def do_something():
  throw SpynlException()

try:
  ...
catch Exception as e:
  do_something() 
```

Even though the exception thrown by do_something() might be completely unrelated to the exception we just caught, it would still be marked as the originial exception.